src/CommonITILValidation.php

```diff
@@ -168,22 +168,71 @@ public function canDeleteItem()
 
 
     /**
-     * Is the current user have right to update the current validation ?
+     * Does the current user have the rights needed to update the current validation?
      *
      * @return boolean
      */
     public function canUpdateItem()
     {
-
+        $is_target = $this->isCurrentUserValidationTarget();
         if (
-            !Session::haveRightsOr(static::$rightname, static::getCreateRights())
-            && ($this->fields["users_id_validate"] != Session::getLoginUserID())
+            !$is_target
+            && !Session::haveRightsOr(static::$rightname, static::getCreateRights())
         ) {
             return false;
         }
         return true;
     }
 
+    /**
+     * Is the validation request for the current user or one or their groups (optional)?
+     * @param bool $include_groups If true, group memberships will be taken into account (true by default)
+     * @return bool
+     */
+    public function isCurrentUserValidationTarget(bool $include_groups = true): bool
+    {
+        $is_target = false;
+        //TODO Remove BC-fix code
+        if (!isset($this->fields['itemtype_target'])) {
+            $this->fields['itemtype_target'] = 'User';
+            $this->fields['items_id_target'] = $this->fields['users_id_validate'];
+        }
```

Comment:
Is this still needed ?

Reply:
The function is public so I can't say how it is used outside the core. The note is for a future version when adding a breaking change is OK.